201-vmss-automation-dsc/nested/provisionNetwork.json

```diff
@@ -0,0 +1,145 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "location": {
+      "type": "string",
+    },
+    "virtualNetworkName": {
+      "type": "string"
+    },
+    "virtualNetworkAddressRange": {
+      "type": "string"
+    },
+    "virtualNetworkSubnets": {
+      "type": "array"
+    },
+    "publicIPAddressName": {
+      "type": "string"
+    },
+    "publicIPAddressType": {
+      "type": "string"
+    },
+    "loadBalancerName": {
+      "type": "string"
+    },
+    "httpProbeRequestPath": {
+      "type": "string",
+      "defaultValue": "/iisstart.htm"
+    }
+  },
+  "resources": [
+    {
+      "name": "[parameters('virtualNetworkName')]",
+      "type": "Microsoft.Network/virtualNetworks",
+      "location": "[parameters('location')]",
+      "apiVersion": "2018-12-01",
+      "properties": {
+        "addressSpace": {
+          "addressPrefixes": [
+            "[parameters('virtualNetworkAddressRange')]"
+          ]
+        },
+        "subnets": "[parameters('virtualNetworkSubnets')]"
+      }
+    },
+    {
+      "name": "[parameters('publicIPAddressName')]",
+      "type": "Microsoft.Network/publicIPAddresses",
+      "location": "[parameters('location')]",
+      "apiVersion": "2018-12-01",
+      "dependsOn": [
+        "[parameters('virtualNetworkName')]"
+      ],
+      "properties": {
+        "publicIPAllocationMethod": "[parameters('publicIPAddressType')]"
+      }
+    },
+    {
+      "type": "Microsoft.Network/loadBalancers",
+      "name": "[parameters('loadBalancerName')]",
+      "location": "[parameters('location')]",
+      "apiVersion": "2018-12-01",
+      "dependsOn": [
+        "[parameters('publicIPAddressName')]"
+      ],
+      "properties": {
+        "frontendIPConfigurations": [
+          {
+            "name": "frontendIPConfiguration",
+            "properties": {
+              "publicIPAddress": {
+                "id": "[resourceId('Microsoft.Network/publicIPAddresses/', parameters('publicIPAddressName'))]"
+              }
+            }
+          }
+        ],
+        "backendAddressPools": [
+          {
+            "name": "backendAddressPool"
+          }
+        ],
+        "probes": [
+          {
+            "name": "httpProbe",
+            "properties": {
+              "protocol": "Http",
+              "port": 80,
+              "intervalInSeconds": 5,
+              "numberOfProbes": 2,
+              "requestPath": "[parameters('httpProbeRequestPath')]"
+            }
+          }
+        ],
+        "loadBalancingRules": [
+          {
+            "name": "loadBalancingRule",
+            "properties": {
+              "frontendIPConfiguration": {
+                "id": "[concat(resourceId('Microsoft.Network/loadBalancers/',parameters('loadBalancerName')),'/frontendIPConfigurations/frontendIPConfiguration')]"
```

Comment:
remove concat, just use resourceId:

resourceId('Microsoft.Network/loadBalancers/frontendIPConfigurations', parameters('loadBalancerName')), 'frontendIPConfiguration')

Here and below

Reply:
That won't work in this case because the "resource name" is a combination of the load balancer name and the front end IP configuration name.  I tried it as a single function and it failed.

https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-resource#resourceid

Reply:
@bmoore-msft

Reply:
It does work, the code I pasted is for a multi-part resourceId (see the SQL example in the docs) - if that fails, send me the code.